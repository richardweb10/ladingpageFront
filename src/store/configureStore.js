import { createStore, applyMiddleware } from "redux";
import createSagaMiddleware from "redux-saga";
import rootReducer from "../reducers";
import rootSaga from "../sagas";
import { createLogger } from "redux-logger/src";

/*const configureStore = () => {
    const sagaMiddleware = createSagaMiddleware();

    return {
        ...createStore(rootReducer, applyMiddleware(sagaMiddleware)),
        runSaga: sagaMiddleware.run(rootSaga)
    }
};

export default configureStore;
*/

const logger = createLogger({
    collapsed: true
});

const configureStore = initialState => {
    const sagaMiddleware = createSagaMiddleware();

    const getMiddleWare = () => {
        if (process.env.NODE_ENV === "development") {
            return applyMiddleware(sagaMiddleware, logger);
        }
        return applyMiddleware(sagaMiddleware);
    };

    const store = createStore(rootReducer, initialState, getMiddleWare());

    sagaMiddleware.run(rootSaga);

    return store;
};

export default configureStore;
