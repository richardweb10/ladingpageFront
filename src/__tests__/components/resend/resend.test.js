import React from 'react';
import { mount, shallow, render } from 'enzyme';
import configureStore from 'redux-mock-store';
import Resend from '../../../components/resend/index';

describe('reenviar', () => {
  const mockStore = configureStore();
  let state = {
    quote: {
      resend: true,
      error: false,
      isLoading: true
    },
    showFormContact: true
  };

  let name,
    email = null;
  const store = mockStore(state);
  const wrapper = mount(<Resend store={store} stateTest={true} />);
  //wrapper.setState({ show: true });
  //wrapper.setState({ showFormContact: true });

  it('resend email', () => {
    email = wrapper.find('input.email');
    email.simulate('focus');
    email.props().value = 'astridduq@gmail.com';
    expect(email.props().value).toBe('astridduq@gmail.com');
  });

  it('resend nombre', () => {
    name = wrapper.find('input#name');
    name.simulate('focus');
    name.props().value = 'astrid';
    expect(name.props().value).toBe('astrid');
  });

  it('resend generar enlace', () => {
    wrapper
      .find('#name')
      .simulate('change', { target: { name: 'name', value: 'Richard' } });

    wrapper.find('#email').simulate('change', {
      target: { name: 'email', value: 'richard.restrepo10@gmail.com' }
    });

    wrapper.find('#btnGenerar').simulate('click');
    expect(wrapper.find('input#jwtTxt').props().value).not.toEqual('');
  });
});
