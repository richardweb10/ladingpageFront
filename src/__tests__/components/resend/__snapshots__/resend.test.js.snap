// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reenviar renders correctly 1`] = `undefined`;
